### ISSUE TITLE
Shapeit5

### ISSUE THREAD

**Issue by rmn8** [ubccr/software-layer#389]
Download shapeit in https://github.com/odelaneau/shapeit5

**Comment by rmn8** [comment]
Attached is the instruction to download https://odelaneau.github.io/shapeit5/docs/installation/build_from_source/system_requirements

**Comment by dsajdak** [comment]
Note for CCR staff: refer to SHAPEIT4 EB recipes and modify for new version

**Comment by la-mendicino** [comment]
Hi Nam,

I installed SHAPEIT5 using a modified easybuild recipe and I've been using it successfully for the past couple months. Shoot me an email at lucas.mendicino@roswellpark.org and I will send you the recipe (I can't attach here for some reason).

I also have the newest version of the 1000 Genomes+Human Genome Diversity Project data available to use as references in SHAPEIT.

**Comment by tonykew** [comment]
I have build SHAPEIT5 from the version 5.1.1 git tag, however there are some
fixes in the main git ranch that I would like to apply before this moves to internal
testing.

**Comment by tonykew** [comment]
SHAPEIT5 version 5.1.1  (with additional patches) has been published to the ccrsoft/2023.01 software release

```
login1$ module spider shapeit5
----------------------------------------------------------------------------
  shapeit5: shapeit5/5.1.1
----------------------------------------------------------------------------
    Description:
      SHAPEIT5 is a fast and accurate method for estimation of haplotypes
      (aka phasing) for SNP array and high coverage sequencing data. 


    You will need to load all module(s) on any one of the lines below before the "shapeit5/5.1.1" module is available to load.

      gcc/11.2.0  openmpi/4.1.1
[...]
login1$ module load gcc/11.2.0 openmpi/4.1.1 shapeit5/5.1.1
login1$ 
```

This includes phase_common and phase_rare

```
login1$ ls ${EBROOTSHAPEIT5}/bin
just-the-docs  phase_common  simulate  xcftools
ligate         phase_rare    switch    xcftools-0.1.1
login1$ 
login1$ phase_common --help

[SHAPEIT5] phase_common (jointly phase multiple common markers)
  * Author        : Olivier DELANEAU, University of Lausanne
  * Contact       : olivier.delaneau@gmail.com
  * Version       : 5.1.1 / commit = 990ed0d / release = 2023-05-08
  * Run date      : 11/10/2024 - 11:07:33

Basic options:
  --help                                Produce help message
  --seed arg (=15052011)                Seed of the random number generator
  -T [ --thread ] arg (=1)              Number of thread used
  --progress                            Verbose progress percentages on screen 
                                        for longer tasks

Input files:
  -I [ --input ] arg                    Genotypes to be phased in VCF/BCF 
                                        format
  -H [ --reference ] arg                Reference panel of haplotypes in 
                                        VCF/BCF format
  -S [ --scaffold ] arg                 Scaffold of haplotypes in VCF/BCF 
                                        format
  -M [ --map ] arg                      Genetic map
  --pedigree arg                        Pedigree information (kid father 
                                        mother)
  --haploids arg                        List of haploid samples (e.g. chrX for 
                                        males). One sample ID per line.
  -R [ --region ] arg                   Target region

MCMC parameters:
  --mcmc-iterations arg (=5b,1p,1b,1p,1b,1p,5m)
                                        Iteration scheme of the MCMC
  --mcmc-prune arg (=0.999)             Pruning threshold for genotype graphs
  --mcmc-noinit                         Disable phasing initialization by PBWT 
                                        sweep

PBWT parameters:
  --pbwt-modulo arg (=0.10000000000000001)
                                        Storage frequency of PBWT indexes in cM
                                        (i.e. storage every 0.1cM by default)
  --pbwt-depth arg (=4)                 Depth of PBWT indexes to condition on 
                                        (default 4)
  --pbwt-mac arg (=5)                   Minimal Minor Allele Count at which 
                                        PBWT is evaluated (default 5)
  --pbwt-mdr arg (=0.10000000000000001) Maximal Missing Data Rate at which PBWT
                                        is evaluated (default 10%)
  --pbwt-window arg (=4)                Run PBWT selection in windows of this 
                                        size (default 4cM)

HMM parameters:
  --hmm-window arg (=4)                 Minimal size of the phasing window in 
                                        cM (default 4cM)
  --hmm-ne arg (=15000)                 Effective size of the population 
                                        (default 15,000)

FILTER parameters:
  --filter-snp                          Only consider SNPs
  --filter-maf arg (=0)                 Only retains variants with a MAF above 
                                        the specified value, requires AC/AN 
                                        tags

Output files:
  -O [ --output ] arg                   Phased haplotypes output file
  --output-format arg (=bcf)            Output file format
  --log arg                             Log file

login1$ 
```
